src/openforms/submissions/logic/actions.py

```diff
@@ -231,17 +231,19 @@ def eval(
         context: DataMapping,
         log: Callable[[JSONValue], None],
     ) -> DataMapping:
-        # XXX perform_service_fetch is expressed in terms of FormVariables
-        dummy_var = FormVariable(
-            name=self.variable,
-            service_fetch_configuration=ServiceFetchConfiguration.objects.get(
-                pk=self.fetch_config
-            ),
-        )
+        if self.fetch_config:  # the old way
```

Comment:
~~can't we make a data migration so this branch of code can be deleted?~~

... we need to support old version export/import... Is there a test for old format imports & ensuring service fetch still works? I'd prefer having the compatibility layer in the import code/serializer rather than "runtime" code if that's possible (but that should be a separate issue/PR).

Reply:
Yeah, fixing this cleanly felt like a rabbit hole and I went for the `if`. I didn't expect codecov to find the `else` path uncovered. I remember changing the code to pass tests and leaving the `if` because of exports and databases containing the old scheme. I've already linked  #2884  to this for a testing a similar "old style" use.

Reply:
Can you then create an issue to move this shim to the import code/serializer and add a `FIXME` with a link to the issue?